client/allocrunner/taskrunner/task_runner_test.go

```diff
@@ -997,6 +997,323 @@ func TestTaskRunner_Download_Retries(t *testing.T) {
+		require.Fail(t, "timed out waiting for task runner to exit")
+	}
+
+	state := tr.TaskState()
+	require.Equal(t, structs.TaskStateDead, state.State)
+	require.True(t, state.Successful())
+	require.False(t, state.Failed)
+
+	artifactsDownloaded := false
+	for _, e := range state.Events {
+		if e.Type == structs.TaskDownloadingArtifacts {
+			artifactsDownloaded = true
+		}
+	}
+	assert.True(t, artifactsDownloaded, "expected artifacts downloaded events")
+
+	// Check that both files exist.
+	_, err = os.Stat(filepath.Join(conf.TaskDir.Dir, f1))
+	require.NoErrorf(t, err, "%v not downloaded", f1)
+
+	_, err = os.Stat(filepath.Join(conf.TaskDir.LocalDir, f2))
+	require.NoErrorf(t, err, "%v not rendered", f2)
+}
+
+func TestTaskRunner_Template_NewVaultToken(t *testing.T) {
+	t.Parallel()
+
+	alloc := mock.BatchAlloc()
+	task := alloc.Job.TaskGroups[0].Tasks[0]
+	task.Templates = []*structs.Template{
+		{
+			EmbeddedTmpl: `{{key "foo"}}`,
+			DestPath:     "local/test",
+			ChangeMode:   structs.TemplateChangeModeNoop,
+		},
+	}
+	task.Vault = &structs.Vault{Policies: []string{"default"}}
+
+	conf, cleanup := testTaskRunnerConfig(t, alloc, task.Name)
+	defer cleanup()
+
+	tr, err := NewTaskRunner(conf)
+	require.NoError(t, err)
+	defer tr.Kill(context.Background(), structs.NewTaskEvent("cleanup"))
+	go tr.Run()
+
+	// Wait for a Vault token
+	var token string
+	testutil.WaitForResult(func() (bool, error) {
+		tr.vaultTokenLock.Lock()
+		defer tr.vaultTokenLock.Unlock()
+
+		token = tr.vaultToken
+
+		if token == "" {
+			return false, fmt.Errorf("No Vault token")
+		}
+
+		return true, nil
+	}, func(err error) {
+		require.NoError(t, err)
+	})
+
+	vault := conf.Vault.(*vaultclient.MockVaultClient)
+	renewalCh, ok := vault.RenewTokens()[token]
+	require.True(t, ok, "no renewal channel for token")
+
+	renewalCh <- fmt.Errorf("Test killing")
+	close(renewalCh)
+
+	var token2 string
+	testutil.WaitForResult(func() (bool, error) {
+		tr.vaultTokenLock.Lock()
+		defer tr.vaultTokenLock.Unlock()
+
+		token2 = tr.vaultToken
+
+		if token2 == "" {
+			return false, fmt.Errorf("No Vault token")
+		}
+
+		if token2 == token {
+			return false, fmt.Errorf("token wasn't recreated")
+		}
+
+		return true, nil
+	}, func(err error) {
+		require.NoError(t, err)
+	})
+
+	// Check the token was revoked
+	testutil.WaitForResult(func() (bool, error) {
+		if len(vault.StoppedTokens()) != 1 {
+			return false, fmt.Errorf("Expected a stopped token: %v", vault.StoppedTokens())
+		}
+
+		if a := vault.StoppedTokens()[0]; a != token {
+			return false, fmt.Errorf("got stopped token %q; want %q", a, token)
+		}
+
+		return true, nil
+	}, func(err error) {
+		require.NoError(t, err)
+	})
+
+}
+
+func TestTaskRunner_VaultManager_Restart(t *testing.T) {
+	t.Parallel()
+
+	alloc := mock.BatchAlloc()
+	task := alloc.Job.TaskGroups[0].Tasks[0]
+	task.Driver = "mock_driver"
+	task.Config = map[string]interface{}{
+		"exit_code": "0",
```

Comment:
This is needed because we want to use 10s time, and we'd need to respecify the whole map.  In this test and the next - but I can remove the Driver line.

Reply:
It will default to 0 if unspecified.